### ISSUE TITLE
proposal: require experimental tag for new rules

### ISSUE THREAD

**Issue by ssbarnea** [ansible/ansible-lint#1031]
##### Summary

1. Newly introduced rules should always have `experimental` tag added to them.
2. The experimental tag is to be removed when a major version is released
3. A major release would never be made if not more than two weeks passed since all experimental new rules were released, giving a window of at least two weeks between a new rule becomes not experimental.

##### Issue Type

- Feature Idea

##### Additional Information

By combining the 3 proposed changes a user can express its preferences like below:

```yaml
# .ansiblelint
skip_list:
  - experimental  # experimental rule violations are skipped
```

```yaml
# .ansiblelint
warn_list:
  - experimental  # experimental rule violations are displayed but do not affect the result
```

It should be mentioned that the default behavior still remains to have all rules enabled. This is an important aspect to remember because it assures that new users will be able to identify problems and is also compatible with use of custom rules. could 

If we implement this we also enable users to also disable any set of tags, for example they could easily add `galaxy` tag to ignore all current and future rules that are related to galaxy.

**Comment by geerlingguy** [comment]
I would be in favor of this but would like to make sure the default behavior is equivalent to:

```
warn_list:
  - experimental
```

That way while a rule is still 'baking', I would start seeing the warnings in output, but they wouldn't immediately kill all 100+ of my CI jobs that run either daily or weekly.

**Comment by ssbarnea** [comment]
I am concerned that a default warning may confuse users but I lets see what other thing about your proposal. If we decide to do it like this it would mean that as soon you put any value in warn_list in the config, the default is overridden. So if someone wants a warn as errors behavior, they would have to write `warn_list: []`
